crates/oxc_linter/src/jest_ast_util.rs

```diff
@@ -0,0 +1,224 @@
+
+pub enum JestFnKind {
+    Hook,
+    Describe,
+    Test,
+    Expect,
+    Jest,
+    Unknown,
+}
+
+impl JestFnKind {
+    pub fn from(name: &str) -> Self {
+        match name {
+            "expect" => Self::Expect,
+            "jest" => Self::Jest,
+            "describe" | "fdescribe" | "xdescribe" => Self::Describe,
+            "fit" | "it" | "test" | "xit" | "xtest" => Self::Test,
+            "beforeAll" | "beforeEach" | "afterAll" | "afterEach" => Self::Hook,
+            _ => Self::Unknown,
+        }
+    }
+}
+
+pub struct ParsedJestFnCall<'a> {
+    pub kind: JestFnKind,
+    pub members: Vec<Cow<'a, str>>,
+    pub raw: Cow<'a, str>,
+}
+
+pub fn parse_jest_fn_call<'a>(
+    call_expr: &'a CallExpression,
+    ctx: &'a LintContext,
+) -> Option<ParsedJestFnCall<'a>> {
+    let callee = &call_expr.callee;
+
+    let resolved = resolve_to_jest_fn(call_expr, ctx);
+
+    // if bailed out, we're not a jest function
+    resolved.as_ref()?;
+
+    let chain = get_node_chain(callee);
+    if let Some(first) = chain.first() && let Some(last) = chain.last() {
+        // if we're an `each()`, ensure we're the outer CallExpression (i.e `.each()()`)
+        if last == "each"
+            && !matches!( callee, Expression::CallExpression(_) | Expression::TaggedTemplateExpression(_))
+        {
+            return None;
+        }
+
+        if matches!(callee, Expression::TaggedTemplateExpression(_)) && last != "each" {
+            return None;
+        }
+
+        let kind = JestFnKind::from(first);
+        let mut members = Vec::new();
+        let mut iter = chain.into_iter();
+        let first = iter.next().expect("first ident name");
+        let rest = iter;
+
+        // every member node must have a member expression as their parent
+        // in order to be part of the call chain we're parsing
+        for member in rest {
+            members.push(member);
+        }
+
+        if let Some(resolved) = resolved {
+            let name = resolved.local;
+            let is_valid_jest_call = if members.is_empty() {
+                VALID_JEST_FN_CALL_CHAINS.iter().any(|chain| chain[0] == name)
+            } else if members.len() == 1 {
+                VALID_JEST_FN_CALL_CHAINS_2.iter().any(|chain| chain[0] == name && chain[1] == members[0])
+            } else if members.len() == 2 {
+                VALID_JEST_FN_CALL_CHAINS_3.iter().any(|chain| chain[0] == name && chain[1] == members[0] && chain[2] == members[1])
+            } else if members.len() == 3 {
+                VALID_JEST_FN_CALL_CHAINS_4.iter().any(|chain| chain[0] == name && chain[1] == members[0] && chain[2] == members[1] && chain[3] == members[2])
+            } else { false };
+
+            if !is_valid_jest_call {
+                return None;
+            }
+        }
+
+        return Some(ParsedJestFnCall { kind, members, raw: first });
+    }
+
+    None
+}
+
+struct ResolvedJestFn<'a> {
+    pub local: Cow<'a, str>,
```

Comment:
why do we need a cow? I don't see anything that's being owned. If everything comes from an Atom, you can also use `&'a Atom`